src/juju/api.ts

```diff
@@ -247,15 +248,22 @@ export async function fetchModelStatus(
         useIdentityProvider,
       );
       if (isLoggedIn(getState(), wsControllerURL)) {
-        status =
-          (await conn?.facades.client?.fullStatus({ patterns: [] })) ?? null;
-        if (!status || typeof status === "string") {
+        try {
+          status =
+            (await conn?.facades.client?.fullStatus({ patterns: [] })) ?? null;
+          if (!status) {
+            // Placeholder error to be caught in the subsequent catch.
+            throw new Error();
```

Comment:
This could be:

```suggestion
            throw new Error("Status not returned.");
```

Reply:
Makes sense to add this. Done! Also removed the comment on line 255 as it becomes redundant.